app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -8,7 +8,7 @@ class ManageIQ::Providers::Autosde::StorageManager::CloudVolume < ::CloudVolume
   def self.raw_create_volume(ext_management_system, options = {})
     # @type [StorageService]
     vol_to_create = ext_management_system.autosde_client.VolumeCreate(
-      :service => options["storage_service"],
+      :service => ExtManagementSystem.find(options["ems_id"]).storage_services.detect { |e| e.id.to_s == options["storage_service_id"] }.ems_ref,
```

Comment:
If you're passed in the `ext_management_system` then you don't need to `ExtManagementSystem.find(options["ems_id"])`

Reply:
> `.storage_services.detect { |e| e.id.to_s == options["storage_service_id"] }.ems_ref`

This looks like it can just be `ext_management_system.storage_services.find_by(:id => options["storage_service_id"])&.ems_ref`

Reply:
done